libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1529,25 +1527,33 @@ void PbftManager::finalize_(SyncBlock &&sync_block, std::vector<h256> &&finalize
         // Update proposal period DAG levels map
         auto ptr = weak_ptr.lock();
         if (!ptr) return;  // it was destroyed
+
+        if (!anchor_hash) {
+          // Null anchor don't update proposal period DAG levels map
+          return;
+        }
+
         auto anchor = dag_blk_mgr_->getDagBlock(anchor_hash);
         if (!anchor) {
           LOG(log_er_) << "DB corrupted - Cannot find anchor block: " << anchor_hash << " in DB.";
           assert(false);
         }
+
         auto new_proposal_period_levels_map = dag_blk_mgr_->newProposePeriodDagLevelsMap(anchor->getLevel());
         db_->addProposalPeriodDagLevelsMapToBatch(*new_proposal_period_levels_map, batch);
         auto dpos_current_max_proposal_period = dag_blk_mgr_->getCurrentMaxProposalPeriod();
         db_->addDposProposalPeriodLevelsFieldToBatch(DposProposalPeriodLevelsStatus::MaxProposalPeriod,
                                                      dpos_current_max_proposal_period, batch);
       });
+
   if (sync) {
     result.wait();
   }
 }
 
 bool PbftManager::pushPbftBlock_(SyncBlock &&sync_block, vec_blk_t &&dag_blocks_order) {
   auto const &pbft_block_hash = sync_block.pbft_blk->getBlockHash();
-  if (db_->pbftBlockInDb(pbft_block_hash)) {
+  if (pbft_chain_->getPbftChainSize() && db_->pbftBlockInDb(pbft_block_hash)) {
```

Comment:
what is this check for ?

Reply:
Check PBFT chain is not empty. In initialization, pbft chain is empty and I hit crash at db_->pbftBlockInDb

Reply:
Check PBFT chain is not empty. If Pbft chain doesn't have any blocks, should not check pbftBlockInDb that will check if the pbft_block_hash is in chain already

Reply:
so when does pbft chain not have any blocks except the very start of the node ?

Reply:
This is for skipping the first block check when first start of a node

Reply:
I know, but it does not make much sense. `pbft_chain_->getPbftChainSize()` is going to be checked thousands of times every day, but the only case when it helps is at the very first start of the node.... After that it is just useless code that needs to be checked and does not bring any value because if always returns value greater than 0

Reply:
Do you have any idea how we can instead of the check?

Reply:
Remove pbft_chain_->getPbftChainSize(). pbftBlockInDb doesn't need check if pbft chain empty.